auth_server/src/auth.rs

```diff
@@ -0,0 +1,40 @@
+use actix_web::{
+    dev::ServiceRequest, error::ErrorUnauthorized, http::header::AUTHORIZATION, Error,
+};
+use jsonwebtoken::{decode, DecodingKey, Validation};
+
+pub struct AuthMiddleware;
+
+impl AuthMiddleware {
+    pub fn validate_token(
+        req: ServiceRequest,
+        secret: &str,
+    ) -> Result<ServiceRequest, (Error, ServiceRequest)> {
+        println!("middleware");
+        dbg!(req.headers().get(AUTHORIZATION));
+        let auth_header = if let Some(header) = req.headers().get(AUTHORIZATION) {
+            if let Ok(auth_str) = header.to_str() {
+                auth_str
+            } else {
+                return Err((ErrorUnauthorized("Invalid authorization header"), req));
+            }
+        } else {
+            return Err((ErrorUnauthorized("Missing authorization header"), req));
+        };
+
+        if !auth_header.starts_with("Bearer ") {
+            return Err((ErrorUnauthorized("Invalid authorization scheme"), req));
+        }
+
+        let token = &auth_header[7..];
+
+        match decode::<super::Claims>(
```

Comment:
It would be better to put this inside the JWT module and not import the jsonwebtoken library anywhere else.